src/libraries/System.Text.Json/gen/Reflection/PropertyInfoWrapper.cs

```diff
@@ -31,6 +32,33 @@ public PropertyInfoWrapper(IPropertySymbol property, MetadataLoadContextInternal
 
         public override string Name => _property.Name;
 
+        /// <summary>
+        /// The exact name specified in the source code. This might be different
+        /// from the <see cref="ClrName"/> because source code might be decorated
+        /// with '@' for reserved keywords, e.g. public string @event { get; set; }
+        /// </summary>
+        public string ExactNameSpecifiedInSourceCode
+        {
+            get
+            {
+                if (_property.DeclaringSyntaxReferences.Length == 1)
+                {
+                    var location = _property.DeclaringSyntaxReferences[0].GetSyntax()?.GetLocation();
+
+                    if (location != null && location != Location.None)
+                    {
+                        var node = location.SourceTree?.GetRoot()?.FindNode(location.SourceSpan) as PropertyDeclarationSyntax;
```

Comment:
Maybe better to use same approach used in the logging generator: https://github.com/dotnet/runtime/blob/09e43903db12547cbecc4be047b28faec9127d74/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs#L310-L317

Reply:
Thanks @am11 - I tried that earlier on, but had all sort of compilation errors. I think something else was screwed, because I've just tried this approach again, and now it works.

Reply:
@am11 - FYI - I changed this to use:
```csharp
NeedsAtSign = SyntaxFacts.GetKeywordKind(_property.Name) != SyntaxKind.None || SyntaxFacts.GetContextualKeywordKind(_property.Name) != SyntaxKind.None
```

(based on a Discord thread that I can no longer find, but I think the suggestion was from @333fred )

Reply:
Nice. We can use the same technique in logging generator (to bring both on the same plan).